components/reddit/sources/new-hot-posts-on-a-subreddit/new-hot-posts-on-a-subreddit.mjs

```diff
@@ -46,13 +46,26 @@ export default {
   },
   hooks: {
     async deploy() {
+      await this.fetchData(10);
+    },
+  },
+  methods: {
+    ...common.methods,
+    generateEventMetadata(redditEvent) {
+      return {
+        id: redditEvent.data.name,
+        summary: redditEvent.data.title,
+        ts: redditEvent.data.created,
+      };
+    },
+    async fetchData(limit) {
       // Emits sample events on the first run during deploy.
       var redditHotPosts = await this.reddit.getNewHotSubredditPosts(
         get(this.subreddit, "value", this.subreddit),
         this.region,
         this.excludeFilters,
         this.includeSubredditDetails,
-        10,
+        limit,
```

Comment:
I think this should be `10` rather than a `limit` variable because this source emits a "new event each time a new hot post is added to the top **10** items in a subreddit".